contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -413,7 +415,7 @@ contract WithdrawalQueue {
             shares -= queue[_firstId - 1].cumulativeShares;
         }
 
-        eth = _min(eth, _toUint128((shares * _shareRate) / 1e9));
+        eth = _min(eth, _toUint128(shares * _shareRate / 1e9));
```

Comment:
maybe 1e27 since `sharesRate` has 1e27 precision multiplier?

Reply:
Nope, here we're converting `shareRate` with 1e27 precision to `eth` with 1e18 precision. So, we have to get rid of additional 1e9. Will add a comment.

Reply:
Ok. Here you are right because we have shares with 1e18 precision too, so we need to reduce 1e27 of zeroes. Thanks. Thats a catch.